src/main/java/com/bookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.bookstore.service.impl;
+
+import com.bookstore.dto.UserRegistrationRequestDto;
+import com.bookstore.dto.UserResponseDto;
+import com.bookstore.exception.RegistrationException;
+import com.bookstore.mapper.UserMapper;
+import com.bookstore.model.User;
+import com.bookstore.repository.UserRepository;
+import com.bookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    
+    private final UserMapper userMapper;
+    
+    @Override
+    public UserResponseDto registerUser(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {

```

Comment:
it's better to use `existsByEmail` to avoid calling `isPresent()`
just add `existsByEmail` to `UserRepository`
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```